### PR TITLE
lossy utf8 conversion on diff read

### PR DESCRIPTION
**Pull request by krlvi**

### CODE REVIEW

**Comment by Qix-** [comment]
@krlvi are we sure that this content isn't written back out to the disk anywhere?

File: crates/gitbutler-core/src/git/diff.rs [hunk 1]
```diff
@@ -191,20 +191,14 @@ fn hunks_by_filepath(
...
                 'B' => assume_binary(),
                 'F' => None,
                 _ => {
-                    if let Ok(content) = str::from_utf8(line.content()) {
-                        Some((content.to_string(), false))
-                    } else {
-                        assume_binary()
-                    }
+                    let content = String::from_utf8_lossy(line.content()).to_string();
+                    Some((content.to_string(), false))
```
Comment (Qix-) [inline crates/gitbutler-core/src/git/diff.rs]:
```suggestion
                    let content = String::from_utf8_lossy(line.content());
                    Some((content.into(), false))
```

Should be sufficient.

Comment (krlvi) [reply crates/gitbutler-core/src/git/diff.rs]: thanks!

**Comment by krlvi** [comment]
> @krlvi are we sure that this content isn't written back out to the disk anywhere?

yeah, actually. i was a bit trigger happy.

**Comment by krlvi** [comment]
I discussed a bit with @Byron earlier - we probably wanna have to have two separate fields on the struct, with explicitly stating that the string bit is display only

**Comment by Qix-** [comment]
What is stored in the non-display field?

**Pull request closed without merging**